aten/src/ATen/CUDAGeneratorImpl.h

```diff
@@ -19,13 +131,26 @@ struct TORCH_CUDA_API CUDAGeneratorImpl : public c10::GeneratorImpl {
   uint64_t seed() override;
   void set_philox_offset_per_thread(uint64_t offset);
   uint64_t philox_offset_per_thread();
+  void graph_prologue(int64_t* offset_extragraph_);
+  uint64_t graph_epilogue();
+  PhiloxCudaState philox_cuda_state(uint64_t increment);
+  void set_is_default(bool is_default) {
+    is_default_ = is_default;
+  }
+
+  // Temporarily accommodates call sites that use philox_engine_inputs.
+  // Allows incremental refactor of call sites to use philox_cuda_state.
   std::pair<uint64_t, uint64_t> philox_engine_inputs(uint64_t increment);
+
   static DeviceType device_type();
 
 private:
   CUDAGeneratorImpl* clone_impl() const override;
   uint64_t seed_ = default_rng_seed_val;
   uint64_t philox_offset_per_thread_ = 0;
+  int64_t* offset_extragraph_;
+  uint32_t offset_intragraph_ = 0;
+  bool is_default_ = false;
```

Comment:
The new `is_default_` field here is expeditious but it is not a good permanent field, since it becomes totally unnecessary when you support local generators in graph capture. So it would be good to make sure people don't sprouting extra uses on it in the meantime. Also, we must be sure that it's not possible to "set" the default CUDA generator (I don't think this is a thing, but with this field it is a more fraught operation).

Hypothetically, you could bypass having this field by simply querying the default CUDA generator and then testing for pointer equality. Might be slower though.

Reply:
I like the pointer comparison better.  "Slower" isn't a problem because the pointer comparison would only trigger if capture is underway, which means it will ultimately be elided.